PlutoHelperAgent/SharedFunctions.m

```diff
@@ -0,0 +1,236 @@
+//
+//  SharedFunctions.m
+//  PlutoHelperAgent
+//
+//  Created by Dave Allison on 05/03/2018.
+//  Copyright © 2018 Guardian News & Media. All rights reserved.
+//
+
+#import "SharedFunctions.h"
+
+@implementation SharedFunctions
+
+NSString *connectionStatusString;
+
++ (NSArray *) load_data_from_keychain {
```

Comment:
Personally I would prefer an NSDictionary here with keys called something like "username", "password", "error" - it's not obvious without going through the code line-by-line that the username is in array[0] and password is in array[1].

@blishen - do you know what Apple's recommendation would be?

Reply:
that way you would not have to return placeholder strings, you could return something in the "error" key of the dict and a caller would then be able to respond accordingly rather than have to check the returned username for string equality

Reply:
Done